src/js/view/controls/controls.js

```diff
@@ -304,6 +321,14 @@ export default class Controls {
                 case 70: // f-key
                     api.setFullscreen();
                     break;
+                case 191: // ? key
+                    if (!this.shortcutsTooltip) {
```

Comment:
shouldn't this check if it is closed ?

Reply:
Nope. It's checking if it exists on the object. If it does, it's already been opened and we can simply toggle the visibility.

Reply:
Also a little confused. on this class `this.shortcutsTooltip` instantiated as null and only updated if we're not on mobile. if the tooltip is falsy, there's no falsy value in JS that will allow you to call a method on it. Are we just not hitting this execution path at all? in just testing?

could also be misreading this. seems like your comment is related to the state of the modal but i don't see anything in this file where the modal state is tied directly to this instance var

Reply:
line 125
` if (!OS.mobile) {
            this.shortcutsTooltip = new ShortcutsTooltip(this.playerContainer, api, model);
        }`

Reply:
yeah so that's when it's instantiated, only on desktop. if we're on mobile though, we're going to hit this part of the code where it'll try to call `this.shortcutsTooltip.open();` on a nonexistent object. is there another place where this is instantiated or is this an execution path we can't hit on mobile? either way i don't know why this block would be needed

Reply:
happy to talk through in person if i'm not making sense!